src/client/automation/playback/move/event-sequence/event-behaviors.js

```diff
@@ -27,12 +27,10 @@ export class MoveBehaviour {
         while (currentParent && currentParent !== commonAncestor) {
             mouseenterElements.push(currentParent);
 
-            currentParent = nativeMethods.nodeParentNodeGetter.call(currentParent);
+            currentParent = domUtils.getParentExceptShadowRoot(currentParent);
         }
 
-        mouseenterElements.reverse();
-
-        for (let i = 0; i < mouseenterElements.length; i++)
+        for (let i = mouseenterElements.length - 1; i > -1; i--)
```

Comment:
(Optional)
We can use `unshift` instead of `push`. In this case we need not to `reverse` the arr or iterate it from the end.

Reply:
I've been thinking about it, but I've decided that the pushing elements to the end of the array is more faster then to the start.

Reply:
It's fast but not clear. Anyway it's ok